File: home_crm_back/src/main/java/ru/dda/homecrmback/domain/subdomain/employee/dto/response/EmployeeDTO.java
package ru.dda.homecrmback.domain.subdomain.employee.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import ru.dda.homecrmback.domain.subdomain.organization.dto.response.OrganizationDTO;
import ru.dda.homecrmback.domain.subdomain.role.dto.response.RoleDTO;
import ru.dda.homecrmback.domain.subdomain.user.dto.response.UserDTO;


@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public record EmployeeDTO(
        long id,
        UserDTO user,
        OrganizationDTO organization,
        RoleDTO role
) {
}

File: home_crm_front/lib/domain/sub/employee/dto/employee_dto.dart
import 'package:home_crm_front/domain/sub/organization/dto/response/organization_dto.dart';
import 'package:home_crm_front/domain/sub/role/dto/role_dto.dart';
import 'package:home_crm_front/domain/sub/user/dto/user_base_dto.dart';
import 'package:json_annotation/json_annotation.dart';

import '../../../support/port/base_dto.dart';
import '../../../support/port/error_data_dto.dart';

part 'employee_dto.g.dart';

@JsonSerializable()
class EmployeeDto {
  final int id;
  final UserBaseDto user;
  final OrganizationDto organization;
  final RoleDto role;

  EmployeeDto({
    required this.id,
    required this.user,
    required this.organization,
    required this.role,
  });

  Map<String, dynamic> toJson() {
    return _$EmployeeDtoToJson(this);
  }

  factory EmployeeDto.fromJson(Map<String, dynamic> json) =>
      _$EmployeeDtoFromJson(json);
}
